pdns/auth-catalogzone.hh

```diff
@@ -41,10 +41,10 @@ public:
     Consumer
   };
 
-  static constexpr const char* getTypeString(enum CatalogType type)
+  static const string& getTypeString(enum CatalogType type)
   {
-    const char* types[] = {"none", "producer", "consumer"};
-    return types[type];
+    static const std::array<const string, 3> types = {"none", "producer", "consumer"};
+    return types.at(type);
```

Comment:
I remember: The main problem of removing the `char*` is loosing the `constexpr` It was one of the reason for trying out `string_view` instead of `string`